lib/spack/spack/parser.py

```diff
@@ -95,13 +96,55 @@
 else:
     FILENAME = WINDOWS_FILENAME
 
+#: These are legal values that *can* be parsed bare, without quotes on the command line.
 VALUE = r"(?:[a-zA-Z_0-9\-+\*.,:=\~\/\\]+)"
-QUOTED_VALUE = r"[\"']+(?:[a-zA-Z_0-9\-+\*.,:=\~\/\\\s]+)[\"']+"
+
+#: Anything that matches this *will* be conservatively quoted in Spack output
+NO_QUOTES_NEEDED = r"^[a-zA-Z0-9,/_.-]+$"
```

Comment:
Does this comment match this value? It looks like if the string matches this it's returned without quotes.

Reply:
nope.  thanks!

Reply:
AFAICT this is the only comment I had that merits a change.  Something like this would do it, happy to approve otherwise.

```suggestion
#: Anything that matches this *will not* be conservatively quoted in Spack output
NO_QUOTES_NEEDED = r"^[a-zA-Z0-9,/_.-]+$"
```

Reply:
pushed -- see what you think